test/createTests.js

```diff
@@ -145,127 +164,134 @@ module.exports = function createTests(createHistory, name, reset = defaultReset)
+          replace: true,
+          state: { bar: 'baz' }
+        });
 
         history.pushState({ bar: 'baz' }, '/bar?query=1#hash=2');
-        expect(store.getState().routing.path).toEqual('/bar?query=1#hash=2');
-        expect(store.getState().routing.replace).toBe(true);
+        expect(store).toContainRoute({
+          path: '/bar?query=1#hash=2',
+          replace: true,
+          state: { bar: 'baz' }
+        });
       });
 
       it('syncs redux -> router', () => {
-        expect(store.getState().routing).toEqual({
+        expect(store).toContainRoute({
           path: '/',
-          changeId: 1,
           replace: false,
           state: undefined
         });
 
         store.dispatch(pushPath('/foo'));
-        expect(store.getState().routing).toEqual({
+        expect(store).toContainRoute({
           path: '/foo',
-          changeId: 2,
           replace: false,
           state: undefined
         });
 
         store.dispatch(pushPath('/foo', { bar: 'baz' }));
-        expect(store.getState().routing).toEqual({
+        expect(store).toContainRoute({
           path: '/foo',
-          changeId: 3,
           replace: false,
           state: { bar: 'baz' }
         });
 
         store.dispatch(replacePath('/bar', { bar: 'foo' }));
-        expect(store.getState().routing).toEqual({
+        expect(store).toContainRoute({
           path: '/bar',
-          changeId: 4,
           replace: true,
           state: { bar: 'foo' }
         });
 
         store.dispatch(pushPath('/bar'));
-        expect(store.getState().routing).toEqual({
+        expect(store).toContainRoute({
           path: '/bar',
-          changeId: 5,
           replace: false,
           state: undefined
         });
 
         store.dispatch(pushPath('/bar?query=1'));
-        expect(store.getState().routing).toEqual({
+        expect(store).toContainRoute({
           path: '/bar?query=1',
-          changeId: 6,
           replace: false,
           state: undefined
         });
 
         store.dispatch(pushPath('/bar?query=1#hash=2'));
-        expect(store.getState().routing).toEqual({
+        expect(store).toContainRoute({
           path: '/bar?query=1#hash=2',
-          changeId: 7,
           replace: false,
           state: undefined
         });
       });
 
       it('updates the router even if path is the same', () => {
-        expect(store.getState().routing).toEqual({
+        expect(store).toContainRoute({
           path: '/',
-          changeId: 1,
           replace: false,
           state: undefined
         });
 
         store.dispatch(pushPath('/foo'));
-        expect(store.getState().routing).toEqual({
+        expect(store).toContainRoute({
           path: '/foo',
-          changeId: 2,
           replace: false,
           state: undefined
         });
 
         store.dispatch(pushPath('/foo'));
-        expect(store.getState().routing).toEqual({
+        expect(store).toContainRoute({
```

Comment:
I think I like this. `changeId` is a case where it might be too much implementation detail to encode in tests.

This is one place where we lose information though. You aren't testing that `history.pushState` is actually called twice.